File: server/src/room_objects/DrawModel.js
"use strict";
//{ "vertices": [{"x": 0.0, "y": 1.0}, ...], "tool": {...}, ...}
var Coordinates = require('./Coordinates');
var DrawObjects = require('./DrawObjects');

var DrawModel = function() {
    this.drawingInformation = new DrawObjects();
    this.listOfCoordinates = [];
    this.thickness = 1;
};

DrawModel.prototype = {
	addCoordinate: function(x, y) {
        var newCoord = new Coordinates(x, y);
        this.listOfCoordinates.push(newCoord);
    },

    setThickness: function(newThickness) {
        if (newThickness >= 0 && newThickness < 100) {
            this.thickness = newThickness;
            return true;
        }

        return false;
    },

    setColour: function(newColour) {
        return this.drawingInformation.setColour(newColour);
    },

    setTool: function(newTool) {
        return this.drawingInformation.setTool(newTool);
    },

    getColour: function(){
        return this.drawingInformation.getColour();
    },

    getTool: function(){
        return this.drawingInformation.getTool();
    },

    getThickness: function(){
        return this.thickness;
    },

    getListOfCoordinates: function(){
        return this.listOfCoordinates;
    },
};

module.exports = DrawModel;

File: client/ios/WhiteboardiOS/WhiteboardiOS/DrawModel.m
//
//  NSObject+DrawModel.m
//  WhiteboardiOS
//
//  Created by Chris Funk on 2015-03-13.
//  Copyright (c) 2015 Patrick Murphy. All rights reserved.
//

#import "DrawModel.h"
#import "Coordinates.h"
#import "DrawObject.h"

@implementation DrawModel

-(id)init
{
    self = [super init];
    
    if (self) {
        _thickness = 1;
        _drawingInformation = [[DrawObject alloc] init];
    }
    
    return self;
}

- (void)addCoordinateX:(NSNumber*)x  Y:(NSNumber*) y {
    NSDictionary *newCord = @{ @"x":x, @"y":y };
    [listOfCoordinates addObject:(newCord)];
}


- (void)setThickness:(NSNumber*)newThickness {
    if (newThickness >= 0 && newThickness < 100) {
        _thickness = newThickness;
    }
}

- (void)SetColour:(NSNumber*)newColour {
    _drawingInformation.setColour(newColour);
}

- (void)setTool:(NSNumber*)newTool {
    _drawingInformation.setTool(newTool);
}
@end